SConstruct

```diff
@@ -1814,6 +1816,13 @@ if env['python_package'] != 'none':
                 f"{python_min_version} or newer is required. In order to install "
                 "Cantera without Python support, specify 'python_package=none'.")
             sys.exit(1)
+    elif python_version >= python_next_version:
+        if env["python_package"] in ("minimal", "full", "default"):
+            logger.error(
+                f"Python version is incompatible. Found {python_version}, which is "
+                f"not yet supported. In order to install Cantera without Python "
+                "support, specify 'python_package=none'.")
+            sys.exit(1)
```

Comment:
I think it makes more sense for this to be a warning. For the scons builder, it's more likely to work by "accident" with new Python versions it seems to me. Plus, if you're building from source, it's implied you have a little more knowledge. The concern to me is more with the pip install crowd getting a bad error message. 

The minimal interface also has a different set of requirements, so I don't think it needs to be included here.

Reply:
:+1: ... I was vacillating between warning and error, and ended up with the latter as it was my assumption that the wheel build may fail when it comes to the Python installation? Still switched to warning as suggested.

Reply:
I guess the CI failure on 3.11.2 when forcing a 'max' version equal to 3.11 answers this question. I'm reverting to what I had before.

Reply:
I updated the error message to indicate that users will have to edit a specific variable in `SConstruct` in case they want to proceed.

Reply:
I still strongly prefer the warning. I think if you do `<=3.11.*` it'll work

Reply:
Sorry, but the build *will* fail, so there is absolutely no point in issuing a warning. Regarding the version string, it’s still easier to stick with 3.12.

Reply:
I still strongly prefer the warning. I think if you do `<=3.11.*` it'll work

Reply:
This unfortunately doesn’t address an attempt to compile with 3.12, which *will* fail due to the setup configuration. For 3.11.*, what is suggested obviously works?

Reply:
Regarding `3.11.*`, it is brittle:
```
>>> from pkg_resources import parse_version
>>> parse_version("3.11.*")
/opt/homebrew/Caskroom/miniforge/base/envs/cantera-dev/lib/python3.10/site-packages/pkg_resources/__init__.py:123: PkgResourcesDeprecationWarning: 3.11.* is an invalid version and will not be supported in a future release
  warnings.warn(
<LegacyVersion('3.11.*')>
```